backend/layers/persistence/persistence_mock.py

```diff
@@ -68,9 +68,7 @@ def create_canonical_collection(
     ) -> CollectionVersion:
         collection_id = CollectionId(self._generate_id())
         version_id = CollectionVersionId(self._generate_id())
-        canonical = CanonicalCollection(
-            id=collection_id, version_id=version_id, originally_published_at=None, tombstoned=False
-        )
+        canonical = CanonicalCollection(collection_id, None, None, False)
```

Comment:
Because the canonical -> version mapping only needs to exist for published collections. A canonical collection can have multiple versions, but only one can be actively published at the same time - hence this mapping. @atolopko-czi has a diagram that might explain this better.

Reply:
what happens if the user goes to an unpublished collection page using the canonical collection id?

Reply:
There is a use case where a user would like to use the canonical id before it is published so they can reference it in their publications.

Reply:
I will add a ticket to make sure this behavior is supported.

Reply:
https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues/gh/chanzuckerberg/single-cell-data-portal/3790